parcels/particlefile.py

```diff
@@ -158,6 +158,8 @@ def write(self, pset, time, sync=True, deleted_only=False):
 
                 self.idx = np.append(self.idx, np.zeros(len(first_write)))
 
+                if np.isnan(pset[0].dt):
```

Comment:
I agree that this should be inside the `for p in pset` call, as it might be that only some particles still have dt set to NaN?

Reply:
Then we shouldn't use warning.
warning_once is not good either since we want the warning each time the export might be not full, right? I'll see if we already have somewhere a warning_once that we can reset after a loop to authorise a next warning again, or do something like this